packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -386,3 +550,687 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
+      color: widget.thumbColor ?? const Color(0x66BCBCBC),
+      thickness: widget.thickness ?? _kScrollbarThickness,
+      fadeoutOpacityAnimation: _fadeoutOpacityAnimation,
+    );
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _maybeTriggerScrollbar();
+  }
+
+  // Waits one frame and cause an empty scroll event (zero delta pixels).
+  //
+  // This allows the thumb to show immediately when isAlwaysShown is true.
+  // A scroll event is required in order to paint the thumb.
+  void _maybeTriggerScrollbar() {
+    WidgetsBinding.instance!.addPostFrameCallback((Duration duration) {
+      if (widget.isAlwaysShown) {
+        _fadeoutTimer?.cancel();
+        // Wait one frame and cause an empty scroll event.  This allows the
+        // thumb to show immediately when isAlwaysShown is true. A scroll
+        // event is required in order to paint the thumb.
+        widget.controller!.position.didUpdateScrollPositionBy(0);
+      }
+    });
+  }
+
+  /// This method is responsible for configuring the [scrollbarPainter]
+  /// according to the [widget]'s properties and any inherited widgets the
+  /// painter depends on, like [Directionality] and [MediaQuery].
+  ///
+  /// Subclasses can override to configure the [scrollbarPainter].
+  @protected
+  void  updateScrollbarPainter() {
+    scrollbarPainter
+      ..color =  widget.thumbColor ?? const Color(0x66BCBCBC)
+      ..textDirection = Directionality.of(context)
+      ..thickness = widget.thickness ?? _kScrollbarThickness
+      ..radius = widget.radius
+      ..padding = MediaQuery.of(context).padding;
+  }
+
+  @override
+  void didUpdateWidget(T oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.isAlwaysShown != oldWidget.isAlwaysShown) {
+      if (widget.isAlwaysShown == true) {
+        _maybeTriggerScrollbar();
+        _fadeoutAnimationController.animateTo(1.0);
+      } else {
+        _fadeoutAnimationController.reverse();
+      }
+    }
+  }
+
+  void _updateScrollPosition(double primaryDelta) {
+    assert(_currentController != null);
+
+    // Convert primaryDelta, the amount that the scrollbar moved since the last
+    // time _dragScrollbar was called, into the coordinate space of the scroll
+    // position, and jump to that position.
+    final double scrollOffsetLocal = scrollbarPainter.getTrackToScroll(primaryDelta);
+    final double scrollOffsetGlobal = scrollOffsetLocal + _currentController!.position.pixels;
+    _currentController!.position.jumpTo(scrollOffsetGlobal);
+  }
+
+  void _maybeStartFadeoutTimer() {
+    if (!widget.isAlwaysShown) {
+      _fadeoutTimer?.cancel();
+      _fadeoutTimer = Timer(widget.timeToFade, () {
+        _fadeoutAnimationController.reverse();
+        _fadeoutTimer = null;
+      });
+    }
+  }
+
+  /// Returns the [Axis] of the child scroll view, or null if one is not
+  /// determined.
+  @protected
+  Axis? getScrollbarDirection() {
+    try {
```

Comment:
This seems pretty strange. Why do we need to ignore all exceptions generated here? Can't we check the _currentController.hasClients before looking up the controller's position?

Reply:
I agree, I have also been curious about it. This was originally part of the `CupertinoScrollbar`. I'll change it to check `_currentController.hasClients` instead.